builtin/checkout.c

```diff
@@ -1064,7 +1176,16 @@ static int switch_branches(const struct checkout_opts *opts,
 			do_merge = 0;
 	}
 
-	if (do_merge) {
+	/* optimize the "checkout -b <new_branch> path */
+	if (!do_merge || skip_merge_working_tree(opts, &old_branch_info, new_branch_info)) {
+		if (!checkout_optimize_new_branch && !opts->quiet) {
+			if (read_cache_preload(NULL) < 0)
+				return error(_("index file corrupt"));
+			show_local_changes(&new_branch_info->commit->object, &opts->diff_options);
+		}
+		if (!opts->only_merge_on_switching_branches)
```

Comment:
For my own understanding, what was the reason for checking `!opts->only_merge_on_switching_branches` here rather than checking `do_merge`?

Reply:
`only_merge_on_switching_branches` is specifically set to 1 in `cmd_switch` and set to 0 in `cmd_checkout`, so its the best way of saying "this is from the `checkout` command, not the `switch` command".